endpoints/openrtb2/auction_test.go

```diff
@@ -120,196 +191,178 @@ func TestExplicitUserId(t *testing.T) {
-			"appnexus": {Disabled: true},
-			"rubicon":  {Disabled: true},
-		},
+
+	aliases, err := jsonparser.GetString(fileData, "mockConfig", "aliases")
+	if err == nil {
+		parsedTestData.Config.AliasJSON = aliases
 	}
-	badTests.assert(t)
-	goodTests.assert(t)
+
+	parsedTestData.Config.BlacklistedAccountArr = parseStringArray(fileData, "blacklistedAccts")
+	parsedTestData.Config.BlacklistedAppArr = parseStringArray(fileData, "blacklistedApps")
+	parsedTestData.Config.AdapterList = parseStringArray(fileData, "disabledAdapters")
+
+	return parsedTestData
 }
 
-// TestBlacklistRequests makes sure we return 400s on blacklisted requests.
-func TestBlacklistRequests(t *testing.T) {
-	// Need to turn off aliases for bad requests as applying the alias can fail on a bad request before the expected error is reached.
-	tests := &getResponseFromDirectory{
-		dir:           "sample-requests/blacklisted",
-		payloadGetter: getRequestPayload,
-		messageGetter: getMessage,
-		expectedCode:  http.StatusServiceUnavailable,
-		aliased:       false,
-	}
-	tests.assert(t)
+func parseStringArray(fileData []byte, jsonField string) []string {
+	rarr := []string{}
+	jsonparser.ArrayEach(fileData, func(value []byte, dataType jsonparser.ValueType, offset int, err error) {
+		rarr = append(rarr, string(value))
+	}, "mockConfig", jsonField)
+	return rarr
 }
 
-// TestRejectAccountRequired asserts we return a 400 code on a request that comes with no user id nor app id
-// if the `AccountRequired` field in the `config.Configuration` structure is set to true
-func TestRejectAccountRequired(t *testing.T) {
-	tests := []*getResponseFromDirectory{
-		{
-			// Account not required and not provided in prebid request
-			dir:           "sample-requests/account-required",
-			file:          "no-acct.json",
-			payloadGetter: getRequestPayload,
-			messageGetter: nilReturner,
-			expectedCode:  http.StatusOK,
-			accountReq:    false,
-		},
-		{
-			// Account was required but not provided in prebid request
-			dir:           "sample-requests/account-required",
-			file:          "no-acct.json",
-			payloadGetter: getRequestPayload,
-			messageGetter: getMessage,
-			expectedCode:  http.StatusBadRequest,
-			accountReq:    true,
-		},
-		{
-			// Account is required, was provided and is not in the blacklisted accounts map
-			dir:           "sample-requests/account-required",
-			file:          "with-acct.json",
-			payloadGetter: getRequestPayload,
-			messageGetter: nilReturner,
-			expectedCode:  http.StatusOK,
-			aliased:       true,
-			accountReq:    true,
-		},
-		{
-			// Account is required, was provided in request and is found in the  blacklisted accounts map
-			dir:           "sample-requests/blacklisted",
-			file:          "blacklisted-acct.json",
-			payloadGetter: getRequestPayload,
-			messageGetter: getMessage,
-			expectedCode:  http.StatusServiceUnavailable,
-			accountReq:    true,
-		},
+func assertTestCaseData(t *testing.T, fileData []byte, testFile string) {
```

Comment:
Since this is both loading the test case and running it, I recommend to name it from `assertTestCaseData` to `runTestCase`. The current names leads me to believe this will just contain the assert statements.